test/tools/unified-spec-runner/runner.ts

```diff
@@ -58,7 +58,9 @@ export async function runUnifiedTest(
   // If test.skipReason is specified, the test runner MUST skip this
   // test and MAY use the string value to log a message.
   if (test.skipReason) {
-    ctx.skipReason = test.skipReason;
+    if (ctx.test) {
```

Comment:
We weren't properly attaching the skip reasons to tests so no skip reason was printed.

Reply:
This I can't answer well - I just followed the typing we have.  Mocha's types have `test` as an optional parameter, and you can't optionally chain it because you can't assign to undefined.

I can't speak to any ramifications of the `test` object being undefined (I imagine that, if the not defined, something has gone wrong with Mocha?  but I don't know for sure.  Maybe it's just lazily initialized) but in this case, we simply wouldn't attach a skip reason.

Reply:
I think you could add this to line 54 the "basic start up" assertions. I think it's optional in Mocha b/c its really mutally exclusive with the currentTest property. 

```ts
expect(ctx.test).to.exist;
```